src/Concordium/Client/Runner.hs

```diff
@@ -1092,13 +1160,13 @@ getDelegatorCooldown bs = do
 
 -- |Query the chain for the given account. Fail if either the chain cannot be reached, or
 -- if the account does not exist.
-getAccountInfoOrDie :: ID.AccountAddress ->  ClientMonad IO Types.AccountInfo
-getAccountInfoOrDie senderAddr = do
-  infoValue <- logFatalOnError =<< withBestBlockHash Nothing (getAccountInfo (Text.pack . show $ senderAddr))
-  case AE.fromJSON $ grpcResponseVal infoValue of
-    AE.Error err -> logFatal ["Cannot decode account info response from the node: " ++ err]
-    AE.Success Nothing -> logFatal [printf "Account %s does not exist on the chain." $ show senderAddr]
-    AE.Success (Just air) -> return air
+getAccountInfoOrDie :: (MonadIO m) => ID.AccountAddress -> BlockHashInput -> ClientMonad m Types.AccountInfo
+getAccountInfoOrDie senderAddr bhInput = do
+  res <- getAccountInfoV2 (Types.AccAddress senderAddr) bhInput
+  dieOnRequestFailed ["I/O error:"] res
+  dieOnStatusNotFound [[i|Account #{senderAddr} does not exist on the chain.|]] res
+  dieOnConversionError ["Cannot decode account info response from the node:"] res
+  getResponseValueOrDie res
```

Comment:
My reasoning for not using the case analysis is that the checks are done in a lot of places, and that I thought it was

1) hard to read (and perhaps quite error prone) as the result datatype is quite complex,
2) the result datatype subject to change (e.g. it has to carry headers whenever the request succeeded at the application level, in a future PR). So if we pattern match everywhere, it think it would be a bit harder to maintain.

This pattern is used a lot in the subsequent PRs, which I will request review on once this is done.

Note that `getResponseValueOrDie` prints an error message as well. The various checks are to keep the error messages as-is as instructed. If you think it is better I can rewrite everything to case expressions, it is like 20 or-so places.